src/app/lib/utilities/preprocessor/rules/timestamp.test.js

```diff
@@ -55,7 +78,106 @@ const headlineBlock = {
   },
 };
 
+const aresMediaBlock = {
+  type: 'aresMedia',
+  model: {
+    blocks: [
+      {
+        type: 'video',
+        model: {
+          blocks: [
+            {
+              type: 'video',
+              model: {
+                text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
+                blocks: [
+                  {
+                    type: 'fragment',
+                    model: {
+                      text:
+                        'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
+                      attributes: [],
+                    },
+                  },
+                ],
+              },
+            },
+          ],
+        },
+      },
+    ],
+  },
+};
+
 describe('Timestamp rules', () => {
+  it('should put Timestamp block after headline and image, if headline exists with image immedeately after it', () => {
+    const fixtureData = {
+      metadata: {
+        firstPublished: 1514808060000,
+        lastPublished: 1514811600000,
+        blockTypes: ['headline', 'text', 'paragraph', 'fragment', 'image'],
+      },
+      content: {
+        model: {
+          blocks: [paragraphBlock, headlineBlock, imageBlock],
+        },
+      },
+    };
+    const expectedTransform = Object.assign(deepClone(fixtureData), {
+      content: {
+        model: {
+          blocks: [
+            paragraphBlock,
+            headlineBlock,
+            imageBlock,
+            {
```

Comment:
This timestamp block can also be pulled out into a constant at the top of the file. e.g. `generatedTimestampBlock`, so it's clear that this is generated within this application, and not via the content management system that's providing the other blocks.